lib/autoproj/github_watcher.rb

```diff
@@ -0,0 +1,168 @@
+require 'autoproj'
+require 'octokit'
+require 'time'
+
+module Autoproj
+    # The GithubWatcher will keep an internal list of watched repositories,
+    # push hooks and pull request hooks. Whenever an event is detected on a
+    # watched repository the corresponding hook is called.
+    class GithubWatcher
```

Comment:
I'd put the daemon-specific code in `autoproj/daemon/` and `Autoproj::Daemon` (it's the gem plugin standard). Only the CLI goes into `Autoproj::CLI`, as it is already structured per-command.

Reply:
Yeah.. I agree but you didn't differently in `autoproj-ci` so I assumed you had a different opinion on that. I will change, what you are suggesting now really is how it should be done